ConsumePlugin/GeneratedCapturingMock.fs

```diff
@@ -0,0 +1,330 @@
+    static member Empty : PrivateTypeInternalFalseMock =
+        {
+            Mem1 = (fun _ -> raise (System.NotImplementedException "Unimplemented mock function: Mem1"))
+            Mem2 = (fun _ -> raise (System.NotImplementedException "Unimplemented mock function: Mem2"))
+            Mem1_Calls = ResizeArray ()
+            Mem2_Calls = ResizeArray ()
+        }
+
+    interface PrivateTypeInternalFalse with
+        member this.Mem1 (arg_0_0, arg_0_1) = this.Mem1 (arg_0_0, arg_0_1)
+        member this.Mem2 arg_0_0 = this.Mem2 (arg_0_0)
+namespace SomeNamespace.CapturingMock
+
+open System
+open WoofWare.Myriad.Plugins
+
+/// Mock record type for an interface
+type internal VeryPublicTypeMock<'a, 'b> =
+    {
+        Mem1 : 'a -> 'b
+        Mem1_Calls : ResizeArray<'a>
+    }
+
+    /// An implementation where every non-unit method throws.
+    static member Empty () : VeryPublicTypeMock<'a, 'b> =
+        {
+            Mem1 = (fun _ -> raise (System.NotImplementedException "Unimplemented mock function: Mem1"))
+            Mem1_Calls = ResizeArray ()
+        }
+
+    interface VeryPublicType<'a, 'b> with
+        member this.Mem1 arg_0_0 = this.Mem1 (arg_0_0)
+namespace SomeNamespace.CapturingMock
+
+open System
+open WoofWare.Myriad.Plugins
+
+/// A single call to the Mem1 method
+type internal Mem1Call<'a> =
+    {
+        arg0 : int
+        arg1 : 'a
+    }
+
+/// A single call to the Mem2 method
+type internal Mem2Call<'a> =
+    {
+        arg0 : int * string
+        arg1 : 'a
+    }
+
+/// A single call to the Mem3 method
+type internal Mem3Call<'a> =
+    {
+        arg0 : int * string
+        arg1 : 'a
+    }
+
+/// A single call to the Mem4 method
+type internal Mem4Call<'a> =
+    {
+        arg0 : int * string
+        arg1 : 'a * int
+    }
+
+/// A single call to the Mem5 method
+type internal Mem5Call<'a> =
+    {
+        arg0 : int * string
+        arg1 : 'a * int
+    }
+
+/// A single call to the Mem6 method
+type internal Mem6Call<'a> =
+    {
+        arg0 : int * string
+        arg1 : 'a * int
+    }
+
+/// Mock record type for an interface
+type internal CurriedMock<'a> =
+    {
+        Mem1 : int -> 'a -> string
+        Mem1_Calls : ResizeArray<Mem1Call<'a>>
+        Mem2 : int * string -> 'a -> string
+        Mem2_Calls : ResizeArray<Mem2Call<'a>>
+        Mem3 : (int * string) -> 'a -> string
+        Mem3_Calls : ResizeArray<Mem3Call<'a>>
+        Mem4 : (int * string) -> ('a * int) -> string
+        Mem4_Calls : ResizeArray<Mem4Call<'a>>
+        Mem5 : int * string -> ('a * int) -> string
+        Mem5_Calls : ResizeArray<Mem5Call<'a>>
+        Mem6 : int * string -> 'a * int -> string
+        Mem6_Calls : ResizeArray<Mem6Call<'a>>
+    }
+
+    /// An implementation where every non-unit method throws.
+    static member Empty () : CurriedMock<'a> =
+        {
+            Mem1 = (fun _ -> raise (System.NotImplementedException "Unimplemented mock function: Mem1"))
```

Comment:
TODO: test that we don't throw for unit-returning methods

Reply:
Actually considering that a won't-fix. The user should specify which methods are implemented.